### PR TITLE
[FLINK-16413]Reduce hive source parallelism when limit push down

### PR DESCRIPTION
**Pull request by zhangjun888**
## What is the purpose of the change

Reduce hive source parallelism when limit push down

## Brief change log

when limit push down ,set the parallelism to min(parallelism,limit)

## Verifying this change

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (100MB)*
  - *Extended integration test for recovery after master (JobManager) failure*
  - *Added test that validates that TaskInfo is transferred only once across recoveries*
  - *Manually verified the change by running a 4 node cluser with 2 JobManagers and 4 TaskManagers, a stateful streaming program, and killing one JobManager and two TaskManagers during the execution, verifying that recovery happens correctly.*

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): no
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: no
  - The serializers: no
  - The runtime per-record code paths (performance sensitive): no
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: no
  - The S3 file system connector:no

## Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/HiveTableSourceTest.java [hunk 1]
```diff
@@ -410,6 +410,32 @@ public void testParallelismSetting() {
...
+		             .commit("pt=1");
+		TableEnvironment tEnv = HiveTestUtils.createTableEnvWithBlinkPlannerBatchMode();
+		tEnv.registerCatalog(catalogName, hiveCatalog);
+		Table table = tEnv.sqlQuery("select * from hive.source_db.test_parallelism_limit_pushdown limit 1");
+		PlannerBase planner = (PlannerBase) ((TableEnvironmentImpl) tEnv).getPlanner();
+		RelNode relNode = planner.optimize(TableTestUtil.toRelNode(table));
+		ExecNode execNode = planner.translateToExecNodePlan(toScala(Collections.singletonList(relNode))).get(0);
+		@SuppressWarnings("unchecked")
+		Transformation transformation = execNode.translateToPlan(planner);
+		Assert.assertEquals(1, transformation.getParallelism());
```
Comment (JingsongLi) [inline flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/HiveTableSourceTest.java]: This is limit transformation, should always be 1.
```
Assert.assertEquals(1, ((PartitionTransformation) ((OneInputTransformation) transformation).getInput())
				.getInput().getParallelism());
```

File: flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableSource.java [hunk 2]
```diff
@@ -168,7 +168,9 @@ public boolean isBounded() {
 			} catch (IOException e) {
 				throw new FlinkHiveException(e);
 			}
-			source.setParallelism(Math.min(Math.max(1, splitNum), max));
+			int parallelism = Math.min(Math.max(1, splitNum), max);
```
[the remark this answers is not in the document]
Comment (zhangjun888) [reply flink-connectors/flink-connector-hive/src/main/java/org/apache/flink/connectors/hive/HiveTableSource.java]: @JingsongLi  Thank you very much for your reply. There are some places that I did not consider carefully. I have modified them according to your suggestions.

**Comment by zhangjun888** [comment]
@JingsongLi  I create a pr for release-1.10 .  https://github.com/apache/flink/pull/11426

**Comment by zhangjun888** [comment]
hi, @JingsongLi  I create a PR for release-1.10    https://github.com/apache/flink/pull/11429

**Pull request merged**